File: 5_10/p085.py
n, x, y = map(int, input().split())
flg = False
for i in range(1, n+1):
    for j in range(i, n+1):
        for k in range(j, n+1):
            s = x - (i + j + k)
            if s <= 0 or s > n:
                continue
            if i * j * k * s == y:
                flg = True
                break
if flg:
    print('Yes')
else:
    print('No')

File: 5_10/p085.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var scan = bufio.NewScanner(os.Stdin)

const (
	initBufSize = 10000
	maxBufSize = 10000000
)

func init() {
	buf := make([]byte, initBufSize)
	scan.Buffer(buf, maxBufSize)
	scan.Split(bufio.ScanWords)
}

func main() {
	n := NumStdin()
	x := NumStdin()
	y := NumStdin()

	ans := false
	for a := 1; a <= n; a++ {
		for b := a; b <= n; b++ {
			for c := b; c <= n; c++ {
				for d := c; d <= n; d++ {
					if a+b+c+d == x && a*b*c*d == y {
						ans = true
						break
					}
				}
			}
		}
	}

	if ans {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

func NumStdin() int {
	scan.Scan()
	num, err := strconv.Atoi(scan.Text())
	if err != nil {
		panic(err)
	}
	return num
}
